plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ResourceMojo.java

```diff
@@ -420,6 +427,16 @@ public void executeInternal() throws MojoExecutionException, MojoFailureExceptio
         }
     }
 
+    private void updateKindFilenameMappings() {
+        if (mappings != null) {
+            final Map<String, List<String>> mappingKindFilename = new HashMap<>();
+            for(MappingConfig mappingConfig : this.mappings) {
+                mappingKindFilename.put(mappingConfig.getKind(), Arrays.asList(mappingConfig.getFilenamesAsArray()));
```

Comment:
You should do a null check here and throw an appropriate error. Remember @Parameter(requied=true) is not evaluated, and hence the kind can be null.